### PR TITLE
Improve Doc Generation

### LINKED ISSUE #1169: Improve generation of objects documentation
Improve the script that generates the objects documentation so that it doesn't adds spaces at the place of the string enumeration if the fields description line is shorter than 40 characters.
See discussion in https://github.com/cyberbotics/webots/pull/1166 about resulting documentation.
An example is the CarpetFibers PROTO.

https://github.com/cyberbotics/webots/blob/a782e2e0df926a3f33edac219a264562b035829b/docs/guide/generate_objects_doc.py#L131-L134

### PR DESCRIPTION
**Pull request by DavidMansolino**
Tasks: 
  - [x] Fix #1169
  - [x] Document automatically field enumeration possibilities.
  - [x] Add links to base nodes doc.

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:+1:

File: docs/guide/object-backgrounds.md [hunk 1]
```diff
@@ -42,11 +53,11 @@ TexturedBackground {
 
 ### TexturedBackground Field Summary
 
-- `texture`: Defines the texture of the background.
+- `texture`: Defines the texture of the background. This field accepts the following values: `"dawn_cloudy_empty"`, `"dusk"`, `"empty_office"`, `"entrance_hall"`, `"factory"`, `"mars"`, `"morning_cloudy_empty"`, `"noon_building_overcast"`, `"noon_cloudy_countryside"`, `"noon_cloudy_empty"`, `"noon_cloudy_mountains"`, `"noon_park_empty"`, `"noon_stormy_empty"`, `"noon_sunny_empty"`, `"noon_sunny_garden"`, `"twilight_cloudy_empty"` and `"mountains"`.
 
 - `skybox`: Specifies if the `texture` field is used to define the skybox shown in the scene background.
 
-- `reflections`: Specifies if the `texture` field is used in the reflections of the [PBRAppearance](../reference/pbrappearance.md) nodes.
+- `reflections`: Specifies if the `texture` field is used in the reflections of the PBRAppearance nodes.
```
Comment (omichel) [inline docs/guide/object-backgrounds.md]: Why did you remove this link?

File: docs/guide/object-road.md [hunk 2]
```diff
@@ -210,7 +210,7 @@ AddLaneRoadSegment {
 
 - `newLaneLeft`: Defines on which side is added the new lane.
 
-- `lines`: Defines the property of each line separating two lanes.
+- `lines`: Defines the property of each line separating two lanes. This field accepts the following value: `RoadLine {}`.
```
Comment (omichel) [inline docs/guide/object-road.md]: This doesn't sound very clear to me. Rather I would say "This filed accepts a list of `RoadLine` nodes".

File: docs/guide/object-road.md [hunk 3]
```diff
@@ -324,7 +324,7 @@ AddLanesRoadSegment {
 
 - `newLaneLeft`: Defines on which side are added the new lanes.
 
-- `lines`: Defines the property of each line separating two lanes.
+- `lines`: Defines the property of each line separating two lanes. This field accepts the following value: `RoadLine {}`.
```
Comment (omichel) [inline docs/guide/object-road.md]: Same comment here (and several other times in this file).

File: docs/guide/object-street-furniture.md [hunk 4]
```diff
@@ -259,7 +259,7 @@ Fence {
 
 - `poleRadius`: Defines the radius of the poles.
 
-- `poleShape`: Defines the shape of the poles. The value should be either `cylinder`, `square`, `diamond` or `H`.
+- `poleShape`: Defines the shape of the poles. The value should be either `cylinder`, `square`, `diamond` or `H`. This field accepts the following values: `"cylinder"`, `"square"`, `"diamond"` and `"H"`.
```
Comment (omichel) [inline docs/guide/object-street-furniture.md]: That seems duplicate.

File: docs/guide/object-trees.md [hunk 5]
```diff
@@ -303,7 +303,7 @@ SimpleTree {
 
 - `name`: Defines the name of the tree.
 
-- `type`: Defines the texture to be used. If set to random, a type will be randomly selected in the list.
+- `type`: Defines the texture to be used. If set to random, a type will be randomly selected in the list. This field accepts the following values: `"random"`, `"oak tree"`, `"crab apple tree"`, `"cherry tree"`, `"birch tree"`, `"palm tree"`, `"spruce"`, `"white pine"`, `"hackberry tree"` and `"hazel tree"`.
```
Comment (omichel) [inline docs/guide/object-trees.md]:
```suggestion
- `type`: Defines the texture to be used. If set to `random`, a type will be randomly selected in the list. This field accepts the following values: `"random"`, `"oak tree"`, `"crab apple tree"`, `"cherry tree"`, `"birch tree"`, `"palm tree"`, `"spruce"`, `"white pine"`, `"hackberry tree"` and `"hazel tree"`.
```

File: docs/guide/object-backgrounds.md (hunk 1 above)
Comment (DavidMansolino) [reply docs/guide/object-backgrounds.md]: Because the link was added manually, which is wrong, we should not edit these files at all as they are generated.
The link can be added in the description of the field in the PROTO itself, but IMO it doesn't really make sense.

Comment (omichel) [reply docs/guide/object-backgrounds.md]: I see. It would be nice however, that our python script generates automatically this kind of links based on a dictionary of nodes.

Comment (DavidMansolino) [reply docs/guide/object-backgrounds.md]: Yes, I will open a new issue about this.

Comment (DavidMansolino) [reply docs/guide/object-backgrounds.md]: Finally, I did add this feature in this PR as it was quite easy, it adds several links previously not present.

**Comment by DavidMansolino** [comment]
:+1:
